docs/trex_tableau_viz.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Add Tableau Viz to your Dashboard Extensions
+layout: docs
+---
+
+Tableau Viz provides a lightweight method for creating visualizations in Tableau. You provide a declarative description of graph or chart, and then call a method to render the description as an SVG image that you can embed in your dashboard extension. The description includes the data (or programmatically adds the data) and specifies the type of chart you wish to create and how it should be encoded. This feature is available through the Dashboard Extensions API and employs the same visualization pipeline that Tableau uses.
+
+
+
+**In this section**
+
+* TOC
+{:toc}
+
+---
+
+## What you need to get started
+
+Tableau Viz is integrated with the Tableau Dashboard Extensions API. To add visualizations to your extension requires the following versions of Tableau and the Dashboard Extensions JavaScript library.
+
+* Tableau 2021.3 and later
+
+* Dashboard Extensions API library (version 1.6 and later)
+
+
+---
+
+
+## The components of a Tableau Viz
+
+There are three main steps in creating a Tableau Viz image.
+
+* Provide the description (or specification) of the image
+
+* Call the Dashboard Extensions method to create the image
+
+* Display the SVG image in the dashboard extension
+
+
+<div class="mermaid">
+graph LR
+   A:[Create input specification] --> B:[Call createVizImageAsync] --> C:[Display SVG in dashboard extension]
+</div>
+
+## Input specification
+
+A Tableau Viz is defined by a specification. This is the `inputSpec` that you pass  as an argument to the `createVizImageAsync` method. The `inputSpec` is a JavaScript object that contains the embedded data and visual specification details. There are two main parts to the `inputSpec`:
+
+* the `data:` (an array of objects, for example, the selected measures and dimensions in the dashboard)
+
+* information about how to format that data (size of viz, mark type, mark color, encoding)
+
+The `inputSpec` is structured in JSON format. The following example shows a specification that creates a simple bar chart. In this example, the data values are statically assigned as part of the specification. In many cases, you would probably programmatically assign the values based upon marks selection in the dashboard or based on some other selection criteria.
```

Comment:
Interesting. I'll change it to mark selection because that sounds better, and is also what extensions API uses (MarkSelectionChanged). Makes sense.  They could just be selecting a single mark.